conda_build/convert.py

```diff
@@ -8,517 +8,701 @@
-            if show:
-                x = match.group(3)
-                print("import", x.replace('/', '.'))
-                matched = True
-            else:
-                return True
-    return matched
+        return '{}{}.{}' .format(build_version,
+            build_version_number[0], build_version_number[1])
 
 
-def has_nonpy_entry_points(t, unix_to_win=True, show=False, quiet=False):
+def extract_temporary_directory(file_path):
+    """Extract the source tar archive contents to a temporary directory.
+
+    Positional arguments:
+    file_path (str) -- the file path to the source package tar file
     """
-    If unix_to_win=True, assumes a Unix type package (i.e., entry points
-    are in the bin directory).
+    temporary_directory = tempfile.mkdtemp()
+
+    source = tarfile.open(file_path)
+    source.extractall(temporary_directory)
+    source.close()
+
+    return temporary_directory
+
+
+def update_dependencies(new_dependencies, existing_dependencies):
+    """Update the source package's existing dependencies.
+
+    When a user passes additional dependencies from the command line,
+    these dependencies will be added to the source package's existing dependencies.
+    If the dependencies passed from the command line are existing dependencies,
+    these existing dependencies are overwritten.
 
-    unix_to_win=False means win to unix, which is not implemented yet, so it
-    will only succeed if there are no entry points.
+    Positional arguments:
+    new_dependencies (List[str]) -- the dependencies passed from the command line
+    existing_dependencies (List[str]) -- the dependencies found in the source
+        package's index.json file
     """
-    if not quiet:
-        print("Checking entry points")
-    bindir = 'bin/' if unix_to_win else 'Scripts/'
-    matched = False
-    for m in t.getmembers():
-        if m.path.startswith(bindir):
-            if not unix_to_win:
-                if show:
-                    print("Entry points with Windows to Unix are not yet " +
-                          "supported")
-                return True
-            r = t.extractfile(m).read()
-            try:
-                r = r.decode('utf-8')
-            except UnicodeDecodeError:
-                if show:
-                    print("Binary file %s" % m.path)
-                matched = True
-            else:
-                firstline = r.splitlines()[0]
-                if 'python' not in firstline:
-                    if show:
-                        print("Non-Python plaintext file %s" % m.path)
-                    matched = True
+    # split dependencies away from their version numbers since we need the names
+    # in order to evaluate duplication
+    dependency_names = set(dependency.split()[0] for dependency in new_dependencies)
+    index_dependency_names = set(index.split()[0] for index in existing_dependencies)
+
+    repeated_packages = index_dependency_names.intersection(dependency_names)
```

Comment:
I believe you could simplify this by converting index_dep_names and dep_names to sets, calling index_dep_names.update(dep_names), and then go back to a list as necessary.